library/core/src/main/java/com/google/android/exoplayer2/util/DurationProvider.java

```diff
@@ -0,0 +1,12 @@
+package com.google.android.exoplayer2.util;
+
+import com.google.android.exoplayer2.C;
+
+public interface DurationProvider {
```

Comment:
There should be no need for this interface or any of the manifest changes as you can access `DownloadHelper.MediaPreparer.timelime`. This timeline already contains the duration of all `Timeline.Period`s.